documentation/src/main/asciidoc/manual-index.asciidoc

```diff
@@ -361,3 +361,271 @@ parameters through - see <<lucene-indexing-performance>> for more details.
+don't provide any batch runtime, you are free to choose one that fits you needs, e.g. the default
+batch runtime embedded in your Java EE container. We provide integration to any CDI-enabled JSR-352
+implementation out of the box. As for other implementations, they can also be used, but requires a
+bit more configuration on the your side.
+
+.How to use the new mass-indexer?
+====
+[source, JAVA]
+----
+Properties jobProps = MassIndexingJob
+        .parameters()
+        .forEntity( MyEntity.class )
+        .build();
+long executionId = BatchRuntime
+        .getJobOperator()
+        .start( MassIndexingJob.NAME, jobProps );
+----
+====
+
+==== Define entities to be indexed
+
+The mass indexing job allows you to define your own entities to be indexed—you can start a full
+indexation or a partial indexation through 3 different methods: selecting the desired entity types,
+using HQL, or using Hibernate criteria.
+
+.How to add define the indexing scope in different methods?
+====
+[source, JAVA]
+----
+// full indexation
+Properties jobProps1 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps2 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( "select c from MyClass c where c.name in ( 'Foo', 'Bar' )" )
+        .build();
+
+// partial indexation using Criteria
+Properties jobProps3 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( Restrictions.in( "name", "Foo", "Bar" ) )
+        .build();
+----
+====
+
+Each method has its pros and cons. The full indexation method allows you to launch a full indexation
+over all the selected entity types, which is very useful when you start using Hibernate Search, or
+when you need to rebuild the indexation after a database migration. However, it's less useful when
+you just want to do partial indexation based on recent changes: it will drop the existing index of
+all the selected entity types. In this case, you need HQL or criteria: they help you to define a
+customized selection, and only the entities inside that selection will be indexed. A typical
+use-case is to index the new entities appeared since yesterday. The inconvenience of such methods is
+they require more configuration. For HQL method, the parallel indexing in is not yet supported.
+
+.Comparaison of Each Method
+|===
+| Method | Indexation | Parallel Indexing | Checkpoint
+
+| Full Indexation
+| Full Indexation
+| Supported
+| Supported
+
+| HQL
+| Partial Indexation
+| Not Supported
+| Not Supported
+
+| Criteria
+| Partial Indexation
+| Supported
+| Supported
+|===
+
+==== Parallel indexing
+
+For better performance, indexation is performed in parallel using multiple threads. The set of
+entities to index is split into multiple partitions, each defined with its own partition ID and
```

Comment:
Yes, I agree with you, Yoann. It's a implementation detail and no need to mention here. Your proposition might lead to misunderstanding, because "element" sounds very similar to the word "item" of JSR-352. So I'll drop this sentence.